### PR TITLE
Implement support for tags in the vm resource

### PR DESCRIPTION
**Pull request by ddelnano**
This PR implements support for #101. Currently the `xenorchestra_vm` resource is implemented but I plan to expand the scope of this to include other resources as well.

## Todo
- [x] Review the code
- [x] ~Add tag support to all the other resources~ - None of the other resources support tags currently. But future resources will have this support by default
- [x] Add docs for the VM resource
- [x] `make testacc` passes

### CODE REVIEW

**Review by dsko1234 — COMMENTED** [review]
This is my first time looking in to the internals of Terraform, so please forgive me if my questions seem stupid.
I am however interested in this feature.

File: xoa/resource_xenorchestra_vm.go [hunk 1]
```diff
@@ -172,6 +172,7 @@ func resourceRecord() *schema.Resource {
 					},
 				},
 			},
+			"tags": resourceTags(),
```
[the remark this answers is not in the document]
Comment (ddelnano) [reply xoa/resource_xenorchestra_vm.go:175]: Tags will be supported on basically every resource so I wanted the logic to be easily reused.

File: xoa/resource_xenorchestra_vm.go [hunk 2]
```diff
@@ -342,19 +351,21 @@ func resourceVmRead(d *schema.ResourceData, m interface{}) error {
...
 
+	id := d.Id()
 	nameLabel := d.Get("name_label").(string)
 	nameDescription := d.Get("name_description").(string)
 	cpus := d.Get("cpus").(int)
 	autoPowerOn := d.Get("auto_poweron").(bool)
 	ha := d.Get("high_availability").(string)
 	rs := d.Get("resource_set").(string)
-	vm, err := c.UpdateVm(d.Id(), cpus, nameLabel, nameDescription, ha, rs, autoPowerOn)
+	vm, err := c.UpdateVm(id, cpus, nameLabel, nameDescription, ha, rs, autoPowerOn)
```
[the remark this answers is not in the document]
Comment (ddelnano) [reply xoa/resource_xenorchestra_vm.go:368]: This change isn't necessary. I thought id needed to be an input to fetch the current tags on a VM but they are already retrieved when the VM strict is read.

[the remark this answers is not in the document]
Comment (ddelnano) [reply xoa/resource_xenorchestra_vm.go:368]: Sorry this is necessary. The code that adds and deletes tags needs the resource id https://github.com/terra-farm/terraform-provider-xenorchestra/pull/107/files#diff-2db9198c1bc5691d1f0c83e9184773676d6efe9d7cc39bf9764c70514cff0213R473

File: docs/resources/vm.md [hunk 3]
```diff
@@ -63,6 +68,7 @@ resource "xenorchestra_vm" "bar" {
     * name_label - (Required) The name for the disk.
     * name_description - (Optional) A description for the disk.
     * size - (Required) The size in bytes of the disk.
+* tags - (Optional) List of labels (strings) that are used to identify and organize resources. These are equivalent to Xenserver [tags](https://docs.citrix.com/en-us/xencenter/7-1/resources-tagging.htmlhttps://docs.citrix.com/en-us/xencenter/7-1/resources-tagging.html).
```
Comment (dsko1234) [inline docs/resources/vm.md:71]:
```suggestion
* tags - (Optional) List of labels (strings) that are used to identify and organize resources. These are equivalent to Xenserver [tags](https://docs.citrix.com/en-us/xencenter/7-1/resources-tagging.html).
```

Comment (ddelnano) [reply docs/resources/vm.md]: Thanks for spotting this 👍

**Pull request merged**